### PR TITLE
feat: use exact due times and remove configurable day start

### LINKED ISSUE #126: feat: use exact due times and remove configurable day start
Determine review eligibility by comparing each FSRS due timestamp with the current time instead of comparing review-day dates. Remove the configurable `dayStartHour` setting and use local midnight for inherently daily features.

Split from #116.

## Implementation

- Use exact timestamps for queue eligibility and same-session requeue decisions.
- Remove `dayStartHour` from settings, storage, import/export, UI, translations, and query invalidation.
- Replace `isDueByDate` in scheduling paths with an exact timestamp comparison.
- Keep review eligibility and calendar bucketing in their existing domain interfaces.
- Adapt the existing domain calendar helpers for local date keys and adjacent-day calculations.
- Use those helpers for daily statistics, streaks, review-history buckets, and upcoming-review labels and buckets.
- Continue accepting legacy exports containing `dayStartHour`, but ignore the field.

## Acceptance criteria

- [ ] A card becomes reviewable only when its FSRS due timestamp is reached.
- [ ] Daily features use local calendar dates beginning at midnight.
- [ ] Scheduling eligibility does not depend on calendar-bucketing helpers.
- [ ] `dayStartHour` is no longer configurable or persisted.
- [ ] Existing user data and legacy imports remain compatible.
- [ ] Tests cover intraday eligibility, local date boundaries, streaks, and chart buckets.

Review policy, delayed-date calculation, and calendar bucketing use the consolidated domain interfaces from #275. Scheduling uses `fsrs.next` directly in the card service (#273). Settings types/validation and statistics types/calculations use the consolidated domain interfaces from #274 and #266. These refactors preserved behavior; exact-time eligibility and day-start removal remain this issue.

### PR DESCRIPTION
**Pull request by mattcdrake**
- Use exact FSRS due times for queue eligibility, requeue decisions, and editor reset; refresh the visible popup queue every 15 seconds.
- Use local midnight for daily limits, statistics, streaks, and chart buckets; remove configurable day start with migration and legacy-import support.
- Remove the mandatory PR screenshot instruction from `AGENTS.md`.
- Verified `npm run check` (970 tests), production build, and calendar tests in Los Angeles and Kolkata timezones.

Closes #126

### CODE REVIEW

File: entrypoints/popup/queries/__tests__/cards.test.tsx [hunk 1]
```diff
@@ -43,6 +45,36 @@ describe('useCardsQuery', () => {
...
+      view.unmount();
+      await act(() => vi.advanceTimersByTimeAsync(30_000));
+      expect(sendMessage).toHaveBeenCalledTimes(2);
+    } finally {
+      view.unmount();
+      focusManager.setFocused(undefined);
+      vi.useRealTimers();
+    }
+  });
+});
```
Comment (mattcdrake) [inline entrypoints/popup/queries/__tests__/cards.test.tsx]: Remove this test.

File: infrastructure/storage/migrations.ts [hunk 2]
```diff
@@ -64,6 +71,9 @@ export async function runStartupMigrations(steps: readonly Migration[] = migrati
       if (migrated.cards !== undefined) {
         await storage.setItem(STORAGE_KEYS.cards, migrated.cards);
       }
+      if (migration.removeKeys) {
+        await storage.removeItems([...migration.removeKeys]);
+      }
```
Comment (mattcdrake) [inline infrastructure/storage/migrations.ts:76]: Add a single test for the added `removeKeys` functionality if none exist.

File: entrypoints/popup/queries/__tests__/cards.test.tsx (hunk 1 above)
Comment (mattcdrake) [reply entrypoints/popup/queries/__tests__/cards.test.tsx]: Removed in 65cbef6. All checks pass (970 tests).

File: infrastructure/storage/migrations.ts (hunk 2 above)
Comment (mattcdrake) [reply infrastructure/storage/migrations.ts:76]: Existing coverage already exercises removeKeys: the migration-v3 test verifies removal while preserving cards, history, and other settings, including a second startup run. The following test covers retry after removal fails. No duplicate test added. See https://github.com/mattcdrake/LeetSRS/blob/65cbef6/infrastructure/storage/__tests__/migrations.test.ts#L331-L368
